base/channels.jl

```diff
@@ -2,6 +2,56 @@
 
 abstract type AbstractChannel end
 
+if JULIA_PARTR
+
+using Base.Threads
+
+"""
+    Channel{T}(sz::Int)
+
+Constructs a `Channel` with an internal buffer that can hold a maximum of `sz` objects
+of type `T`.
+[`put!`](@ref) calls on a full channel block until an object is removed with [`take!`](@ref).
+
+`Channel(0)` constructs an unbuffered channel. `put!` blocks until a matching `take!` is called.
+And vice-versa.
+
+Other constructors:
+
+* `Channel(Inf)`: equivalent to `Channel{Any}(typemax(Int))`
+* `Channel(sz)`: equivalent to `Channel{Any}(sz)`
+"""
+mutable struct Channel{T} <: AbstractChannel
+    cond_take::Condition                 # waiting for data to become available
+    cond_put::Condition                  # waiting for a writeable slot
+    state::Symbol
+    excp::Union{Exception,Nothing}       # exception to be thrown when state != :open
+
+    data::Vector{T}
+    sz_max::Int                          # maximum size of channel
+    lock::SpinLock
+
+    # Used when sz_max == 0, i.e., an unbuffered channel.
+    waiters::Atomic{Int}
```

Comment:
What is used when `sz_max == 0` ? `waiters`? Or the whole block of things after that comment?
(or the lock above)

Should waiters be `n_waiters` or `num_waiters` or `nwaiting` ?
I was confusers seeing it as an `Int` and not as a `Vector{Task}` like `takers` and `putters`
I had to read ahead to workout that the `Int` was a count.
(and idk, not some kind of ID or index into an array or just a typo)